Pokemon/Pokemon/PokemonTableViewController.m

```diff
@@ -0,0 +1,358 @@
+//
+//  PokemonTableViewController.m
+//  Pokemon
+//
+//  Created by Zoufishan Mehdi on 7/31/15.
+//  Copyright (c) 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "PokemonTableViewController.h"
+
+@interface PokemonTableViewController ()
+@property(nonatomic)NSArray *allPokemon;
+@property (nonatomic) NSDictionary *tableData;
+@end
+
+@implementation PokemonTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    
+    self.allPokemon = @[
```

Comment:
You could derive this list from the below (separated) lists, or find some way to represent all the data in one format and then derive the A-Z and type-separated lists from that format. Generally it's a good idea to avoid duplicating data; this is sometimes called DRY or the DRY principle ("Don't Repeat Yourself").

http://c2.com/cgi/wiki?DontRepeatYourself